### ISSUE TITLE
Deploy new CVMFS client test host with CentOS7

### ISSUE THREAD

**Issue by mboisson** [ComputeCanada/software-stack#28]
The current CVMFS clients run CentOS6, which is not supported by the StdEnv/2020 environment. We need to update those.

**Comment by mboisson** [comment]
@sergiykhan, when you have some time, could you help me with that ? cvmfs-client-dev and cvmfs-client currently run CentOS6. We need to redeploy them with CentOS7.

**Comment by sergiykhan** [comment]
Ok, I have just started a new CentOS7 instance (same flavour), did some basic configuration quickly, then took a nap. Will check what specific we have on the current CVMFS clients tomorrow.

**Comment by sergiykhan** [comment]
`cvmfs-client-7.computecanada.ca`
- CentOS7 basic install with most useful packages and sensible configuration
- LDAP authentication and user-dir auto-creation (updated https://wiki.computecanada.ca/staff/LDAP_client_configuration#Linux_authentication with sensible information)
- `/etc/sudoers.d/10-rsnt` is a combination of that from build-node and cvmfs-client. Probably needs a cleanup later.
- `/etc/security/access.conf` is a combination of that from build-node and cvmfs-client. Probably needs a cleanup later.

I have also installed `cvmfs-release`, but do not know what to do next with it. @rptaylor, could you please configure the CVMFS part here to mimic `cvmfs-client.computecanada.ca`?

Once `cvmfs-client-7.computecanada.ca` is finalized, we will make a copy of its volume to spin another instance to mount the dev CVMFS repo.

Once both copies are working, we can swap CentOS6 instances for CentOS7 instance.

**Comment by rptaylor** [comment]
@sergiykhan it is just a matter of installing a few packages following the documentation: https://docs.computecanada.ca/wiki/Accessing_CVMFS#Installing_CVMFS
The default.local config file is a bit more fluid since this node is used for testing, you can just copy whatever the old node has.

**Comment by sergiykhan** [comment]
Ok, I will try that.

**Comment by sergiykhan** [comment]
@rptaylor Ok, so the proxy is `CVMFS_HTTP_PROXY="http://kraken01.westgrid.ca:3128;DIRECT`. I am seeing errors accessing it. Do I have to ask Lixin to allow access to it from my new IP?

**Comment by rptaylor** [comment]
No, it's fine. Besides, I manage that squid.
The CVMFS client is working. I will set up restricted access a different way.

**Comment by sergiykhan** [comment]
Ok, I have consulted with Ryan on the most appropriate CVMFS configuration. It is all done properly now.

@mboisson, could you please take a look at `cvmfs-client-7.computecanada.ca` then to see if you can find anything wrong with it?

**Comment by mboisson** [comment]
It was missing
```/etc/profile.d/z-00-computecanada.sh``` 
I added it. 

It is also missing the user `wiki_module_bot` 

Other than that, I think it is ok.

**Comment by mboisson** [comment]
Ah, it seems to be running on an avx2 host instead of an avx512.

**Comment by mboisson** [comment]
We probably should rsync `/home` between the old and new VM too. 

For `cvmfs-client-dev`, we will need to either do the weird manual CVMFS mount, or maybe automatically use `proot` to overlay it when a new shell is opened.

**Comment by sergiykhan** [comment]
Thanks! I will go though your notes tomorrow.

**Comment by rptaylor** [comment]
I will need the IP of the new dev VM as soon as it is set up, and I will remove the ACL for the IPs of the old VMs when we are done.

**Comment by sergiykhan** [comment]
> I will need the IP of the new dev VM as soon as it is set up, and I will remove the ACL for the IPs of the old VMs when we are done.

Sure. We will keep old and new ones running side by side for a while to make sure we have not missed anything. Then we can decommission the old.

**Comment by sergiykhan** [comment]
`cvmfs-client-7` is running on `avx512` now. The home dirs have been synced. Masao's and Ata's group ownership of their home dirs had wrong GIDs on the original `cvmfs-client`, so I have `chown`ed them before the transfer. I have also reduced the TTLs on the DNS records to 10 min to facilitate the transition. The `wiki_module_bot` user has been added with the same `uid` and `gid` (not sure if it mattered). We can sync home dirs once again before swapping the instances and after announcing the transition to the staff.

Can you tell me more about `cvmfs-client-dev`? The CVMFS configuration there looks essentially identical to that on `cvmfs-client`.

**Comment by ostueker** [comment]
> Can you tell me more about `cvmfs-client-dev`? The CVMFS configuration there looks essentially identical to that on `cvmfs-client`.

The `cvmfs-client-dev` mounts the `/cvmfs/soft-dev.computecanada.ca/` as `/cvmfs/soft.computecanada.ca/`.
This allows us to test software that has been synced to the `dev` repository before syncing it to production.

**Comment by mboisson** [comment]
If I recall correctly, `autofs` is disabled on the dev client, and the CVMFS mounts are done through the `/etc/fstab`.

**Comment by sergiykhan** [comment]
Ok, let me clone `cvmfs-client-7` and then transfer CVMFS configuration and `/etc/fstab` from `cvmfs-client-dev` and see what happens.

**Comment by rptaylor** [comment]
On the dev VM the cvmfs-auto-setup RPM will need to be removed, and the initial autofs setup undone.

**Comment by sergiykhan** [comment]
Alright, could some one, please, take a look at both `cvmfs-client-7` and `cvmfs-client-dev-7`?

Questions:
- I see a cron job for `wiki_module_bot` on both `cvmfs-client` and `cvmfs-client-dev`. I suppose only one of them is valid and needs to be transfered?
- Do you want me to sync home dirs from `cvmfs-client-dev` to `cvmfs-client-dev-7` as well?

**Comment by ostueker** [comment]
> - I see a cron job for `wiki_module_bot` on both `cvmfs-client` and `cvmfs-client-dev`. I suppose only one of them is valid and needs to be transfered?

The `wiki_module_bot` updates the list of Available modules on the (production) docs wiki with modules that have been published to the (production) `/cvmfs/soft.computecanada.ca`.

As I understand, the `wiki_module_bot` on `cvmfs-client-dev` publishes the list of modules installed in `/cvmfs/soft-dev.computecanada.ca` to the doc-dev wiki.

**Comment by mboisson** [comment]
> Alright, could some one, please, take a look at both `cvmfs-client-7` and `cvmfs-client-dev-7`?
> 
> Questions:
> 
> * I see a cron job for `wiki_module_bot` on both `cvmfs-client` and `cvmfs-client-dev`. I suppose only one of them is valid and needs to be transfered?
No, both are valid. They upload to docs and docs-dev, respectively. 
> * Do you want me to sync home dirs from `cvmfs-client-dev` to `cvmfs-client-dev-7` as well?
Yes please.

**Comment by sergiykhan** [comment]
Alright, then I think we are ready to switch over. Here is the plan:

- announce the switch-over in #rnst-software
- update DNS records to point `cvmfs-client` and `cvmfs-client-dev` records to `cvmfs-client-7` and `cvmfs-client-dev-7` (will take 10 min to propagate)
- do a final home dir sync between old and new servers
- comment `wiki_module_bot` cronjobs on old servers, uncomment on the new

Note that current (old) CVMFS clients can still be access as `cvmfs-client-6` and `cvmfs-client-dev-6`. We will keep them around for a couple of weeks as a backup

We can do it today if you want, @mboisson

**Comment by mboisson** [comment]
Yes, it is fine to do it today. Thanks @sergiykhan

**Comment by sergiykhan** [comment]
Ok, it is done. Let's wait for the fallout.

**Comment by mboisson** [comment]
I think this is done. We can destroy the old instances and close this, I would say.

**Comment by sergiykhan** [comment]
Ok, both old instances have been shut-off. Let's wait another week and then:
- terminate instances and their volumes in the cloud
- remove DNS records
- remove LDAP and CVMFS access for those IPs
- rename new CVMFS instances so that there is no `-7` suffix
